airflow-core/tests/unit/executors/test_executor_loader.py

```diff
@@ -560,3 +549,22 @@ def test_single_global_team_configurations_work(self, executor_config, expected_
         with conf_vars({("core", "executor"): executor_config}):
             configs = executor_loader.ExecutorLoader._get_team_executor_configs()
             assert configs == expected_configs
+
+    @pytest.mark.parametrize(
+        "executor_config",
+        [
+            "team1=CeleryExecutor",
+            "team1=CeleryExecutor;team2=LocalExecutor",
+            "team1=CeleryExecutor;team2=LocalExecutor;team3=KubernetesExecutor",
+            "team_a=CeleryExecutor,LocalExecutor;team_b=KubernetesExecutor",
+        ],
+    )
+    def test_team_only_configurations_should_fail(self, executor_config):
+        """Test that configurations with only team-based executors fail validation."""
+        with (
+            mock.patch.object(executor_loader.ExecutorLoader, "block_use_of_multi_team"),
+            mock.patch.object(executor_loader.ExecutorLoader, "_validate_teams_exist_in_database"),
+            conf_vars({("core", "executor"): executor_config}),
+        ):
+            with pytest.raises(AirflowConfigException):
```

Comment:
Nit: Should we match the message too ?
```suggestion
            with pytest.raises(AirflowConfigException, matches=...):
```

Reply:
I actually intentionally didn't add a text matches. Since this is a particular type of exception object I don't think there is much value in matching the text. It mostly is just a nuisance if the message is ever edited. But if you feel strongly I can add one!

Reply:
I ended up adding a small match. Let me know if that suffices